fbw-common/src/wasm/systems/systems/src/communications/receivers.rs

```diff
@@ -0,0 +1,263 @@
+                'J' => ".___",
+                'K' => "_._",
+                'L' => "._..",
+                'M' => "__",
+                'N' => "_.",
+                'O' => "___",
+                'P' => ".__.",
+                'Q' => "__._",
+                'R' => "._.",
+                'S' => "...",
+                'T' => "_",
+                'U' => ".._",
+                'V' => "..._",
+                'W' => ".__",
+                'X' => "_.._",
+                'Y' => "_.__",
+                'Z' => "__..",
+                _ => "",
+            };
+
+            copy.push_str(code);
+            copy.push(' ');
+        }
+
+        copy.chars().rev().collect::<String>()
+    }
+
+    pub fn update(&mut self, context: &UpdateContext) {
+        self.duration_current += context.delta();
+
+        // Manage new ident
+        if self.ident_new != self.ident_current {
+            self.ident_current = self.ident_new;
+            self.morse.clear();
+        }
+
+        // Manage case whenever the morse ident has to restart at the beginning
+        if self.ident_current > 0 && self.morse.is_empty() {
+            self.morse = self.convert_ident_to_morse();
+            self.duration_to_wait = self.duration_end_of_ident;
+            self.duration_current = Duration::from_millis(0);
+        }
+
+        if !self.morse.is_empty() {
+            // If timedout
+            if self.duration_current.as_millis() > self.duration_to_wait as u128 {
+                // After a beep, we have to wait an amount of time equal to a short beep
+                if (self.duration_to_wait == self.duration_short_beep
+                    || self.duration_to_wait == self.duration_long_beep)
+                    && self.beep
+                {
+                    self.duration_to_wait += self.duration_short_beep;
+                    self.beep = false;
+                } else {
+                    self.duration_current = Duration::from_millis(0);
+
+                    match self.morse.pop().unwrap() {
+                        '.' => {
+                            self.duration_to_wait = self.duration_short_beep;
+                            self.beep = true;
+                        }
+                        '_' => {
+                            self.duration_to_wait = self.duration_long_beep;
+                            self.beep = true;
+                        }
+                        _ => {
+                            // space
+                            self.duration_to_wait = self.duration_long_beep;
+                            self.beep = false;
+                        }
+                    };
+                }
+            }
+        } else {
+            self.beep = false;
+        }
+    }
+
+    pub fn get_state(&self) -> bool {
+        self.beep
+    }
+}
+
+impl SimulationElement for Morse {
+    fn read(&mut self, reader: &mut SimulatorReader) {
+        self.ident_new = reader.read(&self.ident_id);
+    }
+}
```

Comment:
This morse beeping would be a lot easier to do in typescript with SoundServer to play packets of sounds (https://microsoft.github.io/msfs-avionics-mirror/docs/sounds/soundserver). Ultimately it needs to be controlled/part of the NAV receivers (in C++) rather than the audio system, so it can be in sync with the simulated NAV radios and ident decoding. We can bring this in now and refactor/rewrite later.

Reply:
Alright sounds good. Always down to create a better code :)

Are the NAV receivers implemented in C++ already? Or are they just planned?